src/atlas/projection/detail/CubedSphereProjectionBase.cc

```diff
@@ -0,0 +1,341 @@
+  ProjectionUtilities::rotate3dZ(angle, xyz);
+}
+
+// -------------------------------------------------------------------------------------------------
+
+void CubedSphereProjectionBase::tile3RotateInverse( double xyz[] ) const {
+  //  Face 3: rotate  90.0 degrees about x axis
+  //          rotate -90.0 degrees about z axis
+  double angle;
+  angle = -M_PI / 2.0;
+  ProjectionUtilities::rotate3dX(angle, xyz);
+  angle = M_PI / 2.0;
+  ProjectionUtilities::rotate3dZ(angle, xyz);
+}
+
+// -------------------------------------------------------------------------------------------------
+
+void CubedSphereProjectionBase::tile4RotateInverse( double xyz[] ) const {
+  //  Face 4: rotate   90.0 degrees about x axis
+  //          rotate -180.0 degrees about z axis
+  double angle;
+  angle = -M_PI / 2.0;
+  ProjectionUtilities::rotate3dX(angle, xyz);
+  angle = M_PI;
+  ProjectionUtilities::rotate3dZ(angle, xyz);
+}
+
+// -------------------------------------------------------------------------------------------------
+
+void CubedSphereProjectionBase::tile5RotateInverse( double xyz[] ) const {
+  //  Face 5: rotate -90.0 degrees about y axis
+  //          rotate -90.0 degrees about z axis
+  double angle;
+  angle = -M_PI / 2.0;
+  ProjectionUtilities::rotate3dY(angle, xyz);
+  angle = -M_PI / 2.0;
+  ProjectionUtilities::rotate3dZ(angle, xyz);
+}
+
+// -------------------------------------------------------------------------------------------------
+
+void CubedSphereProjectionBase::tile6RotateInverse( double xyz[] ) const {
+  //  Face 6: rotate -90.0 degrees about y axis
+  double angle;
+  angle = -M_PI / 2.0;
+  ProjectionUtilities::rotate3dY(angle, xyz);
+}
+
+// -------------------------------------------------------------------------------------------------
+
+void CubedSphereProjectionBase::schmidtTransform( double stretchFac, double targetLon,
+                                                  double targetLat, double lonlat[]) const {
+
+
+  double c2p1 = 1.0 + stretchFac*stretchFac;
+  double c2m1 = 1.0 - stretchFac*stretchFac;
+
+  double sin_p = sin(targetLat);
+  double cos_p = cos(targetLat);
+
+  double sin_lat;
+  double cos_lat;
+  double lat_t;
+
+  if ( std::abs(c2m1) > 1.0e-7 ) {
+    sin_lat = sin(lonlat[LAT]);
+    lat_t = asin( (c2m1+c2p1*sin_lat)/(c2p1+c2m1*sin_lat) );
+  } else {         // no stretching
+    lat_t = lonlat[LAT];
+  }
+
+  sin_lat = sin(lat_t);
+  cos_lat = cos(lat_t);
+  double sin_o = -(sin_p*sin_lat + cos_p*cos_lat*cos(lonlat[LON]));
+
+  if ( (1.-std::abs(sin_o)) < 1.0e-7 ) {    // poles
+    lonlat[LON] = 0.0;
+    lonlat[LAT] = copysign( 0.5*M_PI, sin_o );
+  } else {
+    lonlat[LAT] = asin( sin_o );
+    lonlat[LON] = targetLon + atan2( -cos_lat*sin(lonlat[LON]), -sin_lat*cos_p+cos_lat*sin_p*cos(lonlat[LON]));
+    if ( lonlat[LON] < 0.0 ) {
+      lonlat[LON] = lonlat[LON] + 2.0*M_PI;
+    } else if ( lonlat[LON] >= 2.0*M_PI ) {
+      lonlat[LON] = lonlat[LON] - 2.0*M_PI;
+    }
+  }
+
```

Comment:
I wonder how this algorithm compares to a `RotatedSchmidt` already present in Atlas, which takes parameters:
`double stretching_factor`, and extra parameters for rotation, e.g. `PointLonLat north_pole`, `double rotation_angle`. If it is the same, then that would be handy as it already has the inverse projection as well, which seems to be missing.